earth_enterprise/SConstruct

```diff
@@ -24,6 +26,19 @@ from khEnvironment import khEnvironment
 
 EnsureSConsVersion(0, 96, 92)
 
+# An object used to check for a specific command line parameter.
+# Set the instance variable 'option' based on passed parameter.
+class CommandLineParameter( object ) :
```

Comment:
Style: I don't see this class as checking for a specific command line parameter.  It checks for any parameter that changes the build directory, and which is understood by the `build.gradle` for RPM packaging.

I don't quite understand the need for a class which only has a constructor in it.  In fact, you don't even need a function, you can just evaluate all of this code as part of the `SConstruct` to produce a variable, e.g., something like:

```Python
package_command_arguments = "-Pdebug"

for parameter in [ 'release', 'optimize', 'profile' ]:
    if int( ARGUMENTS.get( parameter, '0' ) ):
        package_command_arguments = "-P" + parameter
        break
```

Reply:
Style: Also the name of this class may have a high likelihood of colliding with the name of anything that processes command-line parameters in general.